tensorflow_io/core/plugins/file_system_plugins.cc

```diff
@@ -17,7 +17,16 @@ limitations under the License.
 
 #include "absl/strings/ascii.h"
 
-void TF_InitPlugin(TF_FilesystemPluginInfo* info) {
+#if defined(_MSC_VER)
+#define TFIO_PLUGIN_EXPORT __declspec(dllexport)
+#else
+#define TFIO_PLUGIN_EXPORT __attribute__((visibility("default")))
+#endif
```

Comment:
@yongtang IIRC, `tensorflow` already has this in `tensorflow/tensorflow/c/c_api_macros.h`. Could you try this out ? Thank you for the PR !